src/Text/Pandoc/Writers/FB2.hs

```diff
@@ -177,20 +179,19 @@ docdate meta' = do
 -- representation.
 renderSections :: PandocMonad m => Int -> [Block] -> FBM m [Content]
 renderSections level blocks = do
-    let secs = splitSections level blocks
-    mapM (renderSection level) secs
-
-renderSection :: PandocMonad m => Int -> ([Inline], [Block]) -> FBM m Content
-renderSection level (ttl, body) = do
-    title <- if null ttl
+    let elements = hierarchicalize blocks
+    cMapM (renderSection level) elements
+
+renderSection :: PandocMonad m =>  Int -> Sh.Element -> FBM m [Content]
+renderSection lvl   (Sh.Blk block) 
+  | lvl == 1 = fmap (list . el "section") (blockToXml block)
```

Comment:
Wraps each 1-level element in section, but we should not have such elements in input file according to FB2 scheme. Original behaviour was to wrap several consequent elements in one section, but I don't know how to implement this.

Reply:
Let me see if I understand the problem.  Suppose hierarchicalize gives you something like
```
Blk, Blk, Blk, Section, Section,...
```
You want to convert this to
```
Section, Section, Section,...
```
where the first Section includes the three initial blocks as content?

Try using the `break` function from Prelude.
```
let isSection Sec{} = True
    isSection _ = False
let (initialBlocks, secs) = break isSection elements
let elements' = Sec 1 [] nullAttr mempty initialBlocks
```
Would something like that work?

Reply:
Small correction:  if `initialBlocks` is null, you wouldn't want to add the extra section.

Reply:
> Let me see if I understand the problem. Suppose hierarchicalize gives you something like
> 
> ```
> Blk, Blk, Blk, Section, Section,...
> ```
> You want to convert this to
> 
> ```
> Section, Section, Section,...

 @jgm Yes, it is now how it works in current Pandoc version before hierarchilize. The question is: should we do such thing at all? According to FB2 specification we can't have `Blk, Blk, Blk, Section, Section,...` on first level, only sections.

Reply:
I'm not sure you understood my proposal.  I understand that the output of hierarchicalize can be
Blk, Blk, Blk, Section, Section, and I also understand that FB2 wants everything in sections.  My proposal was to modify the list of Elements you get back from hierarchicalize, gathering the initial orphan Blks and inserting them into a new Section.

Reply:
@jgm thanks, that's the answer I was waiting for! (which behaviour should be implemented). 
I looked at `break`, but it does the trick only for the first orphaned blocks, like `Blk Blk Section Blk Blk` -> `Secion Secion Blk Blk`. 
Maybe we should use folding to wrap every sequence of blocks? Or maybe there is more powerful function?

Reply:
Thank you for help. All tests are passed now.